src/views/ModelPlan/TaskList/Payment/BeneficiaryCostSharing/index.tsx

```diff
@@ -0,0 +1,452 @@
+          handleFormSubmit(values, 'next');
+        }}
+        enableReinitialize
+        innerRef={formikRef}
+      >
+        {(formikProps: FormikProps<BeneficiaryCostSharingFormType>) => {
+          const {
+            errors,
+            handleSubmit,
+            setFieldValue,
+            setErrors,
+            values
+          } = formikProps;
+          const flatErrors = flattenErrors(errors);
+          return (
+            <>
+              {Object.keys(errors).length > 0 && (
+                <ErrorAlert
+                  testId="formik-validation-errors"
+                  classNames="margin-top-3"
+                  heading={h('checkAndFix')}
+                >
+                  {Object.keys(flatErrors).map(key => {
+                    return (
+                      <ErrorAlertMessage
+                        key={`Error.${key}`}
+                        errorKey={key}
+                        message={flatErrors[key]}
+                      />
+                    );
+                  })}
+                </ErrorAlert>
+              )}
+              <GridContainer className="padding-left-0 padding-right-0">
+                <Grid row gap>
+                  <Grid desktop={{ col: 6 }}>
+                    <Form
+                      className="margin-top-6"
+                      data-testid="payment-beneficiary-cost-sharing-form"
+                      onSubmit={e => {
+                        handleSubmit(e);
+                      }}
+                    >
+                      <PageHeading
+                        headingLevel="h3"
+                        className="margin-bottom-3"
+                      >
+                        {t('beneficaryCostSharingQuestions')}
+                      </PageHeading>
+
+                      <FieldGroup
+                        scrollElement="payment-beneficiary-cost-sharing"
+                        error={
+                          !!flatErrors.beneficiaryCostSharingLevelAndHandling
+                        }
+                      >
+                        <Label
+                          htmlFor="payment-beneficiary-cost-sharing"
+                          className="maxw-none"
+                        >
+                          {t('beneficiaryCostSharingLevelAndHandling')}
+                        </Label>
+                        <FieldErrorMsg>
+                          {flatErrors.beneficiaryCostSharingLevelAndHandling}
+                        </FieldErrorMsg>
+                        <Field
+                          as={TextAreaField}
+                          className="height-15"
+                          error={
+                            flatErrors.beneficiaryCostSharingLevelAndHandling
+                          }
+                          id="payment-beneficiary-cost-sharing"
+                          data-testid="payment-beneficiary-cost-sharing"
+                          name="beneficiaryCostSharingLevelAndHandling"
```

Comment:
All of the TexaAreaField "name" attributes need to be the field key for this page.